mava/networks.py

```diff
@@ -51,19 +52,38 @@ def __call__(self, observation: chex.Array) -> chex.Array:
         return x
 
 
-class FeedForwardActor(nn.Module):
-    """Feedforward Actor Network."""
+class CNNTorso(nn.Module):
+    """CNN torso."""
 
-    torso: nn.Module
-    num_actions: Sequence[int]
+    channel_sizes: Sequence[int]
+    kernel_sizes: Sequence[int]
+    strides: Sequence[int]
+    activation_fn: Callable[[chex.Array], chex.Array] = nn.relu
+    use_layer_norm: bool = False
 
     @nn.compact
-    def __call__(self, observation: Observation) -> distrax.Categorical:
+    def __call__(self, observation: chex.Array) -> chex.Array:
         """Forward pass."""
-        x = observation.agents_view
+        x = observation
+        for channel, kernel, stride in zip(self.channel_sizes, self.kernel_sizes, self.strides):
+            x = nn.Conv(channel, (kernel, kernel), (stride, stride))(x)
+            if self.use_layer_norm:
+                x = nn.LayerNorm(use_scale=False)(x)
+            x = self.activation_fn(x)
 
-        x = self.torso(x)
-        actor_logits = nn.Dense(self.num_actions, kernel_init=orthogonal(0.01))(x)
+        return x.reshape((x.shape[0], -1))
+
+
+class DiscreteActionHead(nn.Module):
+    """Discrete Action Head"""
+
+    action_dim: int
+
+    @nn.compact
+    def __call__(self, x: chex.Array, observation: Observation) -> distrax.Categorical:
```

Comment:
Why pass in x and observation here? I think we should always pass in observation

Reply:
So x is for the hidden layer output of a torso - the only reason observation is passed in is for the action mask and/or if somebody wanted to use additional information from the observation.

Reply:
Could we name it better/add docs to make this clearer as it wasn't clear at first glance

Reply:
I replaced `observation` -> `observation.action_mask` to make things clearer. The action head's input name was also updated. What do you think of this @sash-a?